backend/views/core/clients/create.py

```diff
@@ -25,35 +25,41 @@ def create_client(request: HtmxHttpRequest):
         messages.error(request, error)
         return redirect("clients create")
 
-    if request.user.logged_in_as_team:
-        client = Client.objects.create(
-            organization=request.user.logged_in_as_team,
-        )
+    client = create_client_instance(request.user, client_details)
+
+    if not client:
+        messages.error(request, "Failed to create client - an unknown error occurred")
     else:
-        client = Client.objects.create(
-            user=request.user,
-        )
+        messages.success(request, f"Client created successfully (#{client.id})")
 
-    for model_field, new_value in client_details.items():
-        setattr(client, model_field, new_value)
+    return redirect("clients dashboard")
 
-    client.save()
 
-    if client:
-        messages.success(request, f"Client created successfully (#{client.id})")
-    else:
-        messages.error(request, "Failed to create client - an unknown error occurred")
-    return redirect("clients dashboard")
+def create_client_instance(user, client_details):
+    kwargs = {"organization": user.logged_in_as_team} if user.logged_in_as_team else {"user": user}
+    try:
+        client = Client.objects.create(**kwargs)
```

Comment:
Would it not be more efficient instead of looping through the setattr like that, to just add them where you add kwargs? You can use the union operator to add two dicts. Like `**kwargs | client_details` I believe

Reply:
Ah I see, this was already there before, you just refactored it. Apologies lol, just my bad code :)